bridge/checkpoint/transport/transport.go

```diff
@@ -62,15 +66,36 @@ func (t *libp2pGossipTransport) Subscribe(handler func(interface{})) error {
 
 		// and call handler
 
-		// protoMessage, ok := obj.(*proto.SignedMessage)
-		// if !ok {
-		// 	t.logger.Warn("received unexpected typed message", "message", obj)
+		protoMsg, ok := obj.(*proto.CheckpointMessage)
+		if !ok {
+			t.logger.Warn("received unexpected typed message", "message", obj)
+
+			return
+		}
+
+		//	convert message to appropriate type
+		var message interface{}
+
+		switch protoMsg.Type {
+		case proto.CheckpointMessage_CHECKPOINT:
+			checkpoint := ctypes.Checkpoint{}
+			if err := checkpoint.UnmarshalRLP(protoMsg.Payload); err != nil {
+				t.logger.Error("unable to unmarshal payload from message", "err", err)
+
+				return
+			}
```

Comment:
This unmarshalling can be moved below the switch, so in your case statements you only set the unmarshaller base